src/main/java/com/bisnode/opa/client/OpaClient.java

```diff
@@ -73,7 +75,12 @@ public Builder opaConfiguration(String url) {
         }
 
         public OpaClient build() {
-            HttpClient httpClient = HttpClient.newHttpClient();
+            Objects.requireNonNull(opaConfiguration, "build() called without opaConfiguration provided");
+            HttpClient httpClient = HttpClient.newBuilder()
+                    .version("https".equals(URI.create(opaConfiguration.getUrl()).getScheme()) ?
```

Comment:
Can we move this logic to OpaConfiguration? This way it'll be overridable and by default it will be "automagically" resolved.

Reply:
At this point in time, this seems unimportant enough not to warrant changes to public methods or constructors IMO, but when [this](https://github.com/open-policy-agent/opa/pull/2739) makes it into OPA that could change. I'm not sure how well supported that is by the Java 11 HTTP client though. Docs on the subject seem sparse to say the least.

Moving it to OpaConfiguration without override (at least not at this point in time) though, would that make sense?

Reply:
> Moving it to OpaConfiguration without override (at least not at this point in time) though, would that make sense?

I think this is the best thing to do currently.

Reply:
I made the move, but it felt awkward having a getter with no way of setting it explicitly. So I added a constructor for that case too. We will likely want to review the public APIs on this before 1.0.